integrations/oci/oci-secrets-config-source/src/main/java/io/helidon/integrations/oci/secrets/configsource/OciSecretsConfigSourceProvider.java

```diff
@@ -0,0 +1,526 @@
+ *
+ * <p>More specifically:</p>
+ *
+ * <ol>
+ *
+ * <li>Ensure you have an authentication mechanism set up to connect to OCI (e.g. a valid <a
+ * href="https://docs.oracle.com/en-us/iaas/Content/API/Concepts/sdkconfig.htm">OCI configuration file</a>)</li>
+ *
+ * <li>Ensure there is a classpath resource present named {@code meta-config.yaml}.</li>
+ *
+ * <li>Ensure the {@code meta-config.yaml} classpath resource contains a {@code source} with a {@code type} of {@code
+ * oci-secrets} that looks similar to the following, substituting values as appropriate:<blockquote><pre>sources:
+ *  - type: 'oci-secrets'
+ *    properties:
+ *      compartment-ocid: 'your vault compartment OCID goes here'
+ *      vault-ocid: 'your vault OCID goes here'</pre></blockquote></li>
+ *
+ * </ol>
+ *
+ * <p>Refer to Helidon's documentation concerning meta-configuration for more details.</p>
+ *
+ * @see ConfigSourceProvider
+ */
+public final class OciSecretsConfigSourceProvider implements ConfigSourceProvider, Weighted {
+
+
+    /*
+     * Static fields.
+     */
+
+
+    private static final Logger LOGGER = System.getLogger(OciSecretsConfigSourceProvider.class.getName());
+
+    private static final Set<String> SUPPORTED_TYPES = Set.of("oci-secrets");
+
+    private static final double WEIGHT = 300D;
+
+
+    /*
+     * Constructors.
+     */
+
+
+    /**
+     * Creates a new {@link OciSecretsConfigSourceProvider}.
+     *
+     * @deprecated For use by {@link java.util.ServiceLoader} only.
+     */
+    @Deprecated // For use by java.util.ServiceLoader only.
+    public OciSecretsConfigSourceProvider() {
+        super();
+    }
+
+
+    /*
+     * Instance methods.
+     */
+
+
+    /**
+     * Creates and returns a non-{@code null} {@link ConfigSource} that sources its values from an Oracle Cloud
+     * Infrastructure (OCI) <a
+     * href="https://docs.oracle.com/en-us/iaas/Content/KeyManagement/Concepts/keyoverview.htm">Vault</a>.
+     *
+     * @param type one of the {@linkplain #supported() supported types}, or an {@linkplain ConfigSources#empty() empty
+     * <code>ConfigSource</code>} will be returned
+     *
+     * @param metaConfig a {@link Config} serving as meta-configuration for this provider; must not be {@code null} when
+     * {@code type} is {@linkplain #supports(String) supported}
+     *
+     * @return a non-{@code null} {@link ConfigSource}
+     *
+     * @exception NullPointerException if {@code type} is {@linkplain #supports(String) supported} and {@code
+     * metaConfig} is {@code null}
+     *
+     * @see #supported()
+     *
+     * @deprecated For use by the Helidon Config subsystem only.
+     */
+    @Deprecated // For use by the Helidon Config subsystem only.
+    @Override // ConfigSourceProvider
+    public ConfigSource create(String type, Config metaConfig) {
+        return this.supports(type) ? SecretBundleConfigSource.builder().config(metaConfig).build() : ConfigSources.empty();
```

Comment:
This method is only called for a supported type, you do not need to check it, and you do not need to return an empty config source.
The `type` support is built in for providers that have more than one supported type, so they know which one to build.

Reply:
I will change this method to do no checking as requested.

Reply:
Done